pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -997,6 +997,11 @@ func buildDefaultCluster(env *model.Environment, name string, discoveryType apiv
 		cluster.DnsLookupFamily = apiv2.Cluster_V4_ONLY
 	}
 
+	if discoveryType == apiv2.Cluster_STRICT_DNS {
```

Comment:
or logical dns right? Not that we use logical DNS anywhere. if so, we should remove logical dns from the if block below